db/sqlinterfaces.c

```diff
@@ -5712,80 +5699,67 @@ static void sqlengine_work_lua_thread(struct thdpool *pool, void *work,
 
 int gbl_debug_sqlthd_failures;
 
-static void sqlengine_work_appsock(struct thdpool *pool, void *work,
-                                   void *thddata)
+static int execute_verify_indexes_int(struct sqlthdstate *thd,
+                                  struct sqlclntstate *clnt)
 {
-    struct sqlthdstate *thd = thddata;
-    struct sqlclntstate *clnt = work;
     int rc;
+    if (thd->sqldb == NULL) {
+        if ((rc = sqlengine_prepare_engine(thd, clnt)) != 0) {
+            return rc;
+        }
+    }
+    sqlite3_stmt *stmt;
+    const char *tail;
+    rc = sqlite3_prepare_v2(thd->sqldb, clnt->sql, -1, &stmt, &tail);
+    if (rc != SQLITE_OK) {
+        return rc;
+    }
+    bind_verify_indexes_query(stmt, clnt->schema_mems);
+    run_stmt_setup(clnt, stmt);
+    if ((rc = sqlite3_step(stmt)) == SQLITE_ROW) {
+        clnt->has_sqliterow = 1;
+        return verify_indexes_column_value(stmt, clnt->schema_mems);
+    }
+    clnt->has_sqliterow = 0;
+    if (rc == SQLITE_DONE) {
+        return 0;
+    }
+    return rc;
+}
+
+static int execute_verify_indexes(struct sqlthdstate *thd,
+                                  struct sqlclntstate *clnt)
+{
+    rdlock_schema_lk();
```

Comment:
For `verify_indexes` queries (i.e. partial/functional indexes' internal queries), there is no need to get schemalk or to re-prepare engine. We can just simply run them as these queries can only be common table expressions. see [here](https://github.com/bloomberg/comdb2/pull/387/files#diff-4932981bbf2c2bd6bf0267d1a6d39173L5741)

Reply:
Maybe I should take this whole logic out of sqlinterfaces.c?

Reply:
Perhaps that is a better way to do it.